### ISSUE TITLE
Quick Start with Docker / Docker Compose

### ISSUE THREAD

**Issue by hrryndrsn** [FuelLabs/fuel-indexer#484]
Create a dockerized version of the quick start demo which includes sample project files and dependencies.
This could be a single `Dockerfile` or multiple containers in a `docker-compose.yml` file.

> Clone our sample project and run:
```
docker compose up
```

Inspiration:
- [cardano-wallet](https://github.com/input-output-hk/cardano-wallet)
- [eth-docker](https://eth-docker.net/)

**Comment by hrryndrsn** [comment]
Simple WIP `Dockerfile` with sqlite database
```Dockerfile
FROM ubuntu:latest AS base
ENV DEBIAN_FRONTEND=noninteractive
RUN apt-get update && \
    apt-get upgrade -y && \
    apt-get install -y software-properties-common curl git sudo && \
    apt-get install -y cmake pkg-config gcc build-essential && \
    apt-get install -y clang libclang-dev llvm libpq-dev && \
    apt-get install -y sqlite3 && \
    apt-get clean autoclean && \
    apt-get autoremove --yes

FROM base AS non_root
ARG USERNAME=user
ARG USER_UID=1000
ARG USER_GID=$USER_UID
# Create the user
RUN groupadd --gid $USER_GID $USERNAME \
    && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME 
USER $USERNAME
WORKDIR /home/$USERNAME
#Install Rust Toolchain
RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
ENV PATH "/home/${USERNAME}/.cargo/bin:${PATH}"
RUN cargo -v
# Install Fuel Toolchain
RUN curl --proto '=https' --tlsv1.2 -sSf https://fuellabs.github.io/fuelup/fuelup-init.sh | sh -s -- -y
ENV PATH="/home/${USERNAME}/.fuelup/bin:${PATH}"
# Install sqlx-cli
RUN cargo install sqlx-cli
# Database - SQLite
RUN sqlite3 indexer_database
ENV DATABASE_URL "sqlite://indexer_database.db"

ENTRYPOINT ["/bin/bash"]

```

**Comment by hrryndrsn** [comment]
Stuck on a couple of things:

- forc index build command not found
- paths and url matching in the manifest and lib

**Comment by ra0x3** [comment]
Hey @hrryndrsn I like this idea. It's on theme with some of the feedback we've received regarding making the Quickstart quicker. This seems to be closely related to the work I'm doing in https://github.com/FuelLabs/fuel-indexer/pull/486 so I can just include a `docker-compose.yml` in my PR

**Comment by hrryndrsn** [comment]
@ra0x3 I made a 'Proof of Concept` as a cloneable repo [fuel_indexer_docker_quick_start](https://github.com/hrryndrsn/fuel_indexer_docker_quick_start). Just under my github  just mainly as an example to get feedback.

Yeah if we have an existing docker compose ill look using that!

**Comment by hrryndrsn** [comment]
@ra0x3  Yep I had a look at the compose we already have, i think its a good first step to just use that. Does is bootstrap a fuel node as well?

I think the downside of using a prebuilt container is that we need to add that the CI/CD, where if it's a Dockerfile then people can build it one their own. Its not too much stuff just the system dependencies and our repository examples. 

Really the intention is to provide an example of how these components are used together rather than a production ready setup.

**Comment by hrryndrsn** [comment]
Note to self: Don’t use repo examples as they are configured specifically for the cargo workspace of the indexer project. Use forc index new to bootstrap a new index and deploy it

**Comment by hrryndrsn** [comment]
fixed by #486
